Tasks/RunPowerShellOnTargetMachines/RunPowerShellOnTargetMachines.ps1

```diff
@@ -21,36 +26,130 @@ Write-Verbose "runPowershellInParallel = $runPowershellInParallel" -Verbose
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
 
-$connection = Get-VssConnection -TaskContext $distributedTaskContext
+	# Constants #
+$resourceFQDNKeyName = 'Microsoft-Vslabs-MG-Resource-FQDN'
+$resourceWinRMHttpPortKeyName = 'WinRM_HttpPort'
+$resourceWinRMHttpsPortKeyName = 'WinRM_HttpsPort'
+$httpConnectionProtocolValue = 'http'
+$httpsConnectionProtocolValue = 'https'
+$skipCACheckOptionValue = 'doSkipCAchek'
 
-$port = '5985'
 
-$resources = Get-EnvironmentResources -EnvironmentName $environmentName -ResourceFilter $machineNames -Connection $connection -ErrorAction Stop
+	# Defaults #
+$envOperationStatus = "Passed"
+$defaultWirmHttpPort = '5985'
+$defaultWirmHttpsPort = '5986'
+$useHttpProtocol = $true
+$httpProtocallOption = '-UseHttp'
+$doSkipCACheckOption = '-SkipCACheck'
+
+	# Process input parameters #
+if($connectionProtocol -eq $httpsConnectionProtocolValue)
+{
+	$useHttpProtocol = $false	# use https connection
+	$httpProtocallOption = ''
+	
+	# check if certificate authentication is opted for https connection
+	if($security -ne $skipCACheckOptionValue )
+	{
+		$doSkipCACheckOption = ''
+		Write-Verbose "Certificate authentication will be used to connect to target machines for given winrm https port" -Verbose
```

Comment:
Check the msdn https://technet.microsoft.com/library/hh849703.aspx for explaining skipCA option for newPSSessionOption. Use similar wording in verbose below.